lib/plugins/wallet/cardano/cardano.js

```diff
@@ -0,0 +1,85 @@
+const { walletId } = require('./mnemonic')
+const cardanoApi = require('./client')
+
+// TODO: work with BN here instead of api.js?
+
+async function checkCryptoCode(cryptoCode) {
+  if (cryptoCode !== 'ADA') {
+    return Promise.reject(new Error(`Unsupported crypto: ${cryptoCode}`))
+  }
+
+  return Promise.resolve()
+}
+
+/**
+ * TODO: instead of using walletId function we can
+ *       inject walletId/index into account object
+ *       in wallet.js - it's better?
+ */
+async function balance(account, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId }) => cardanoApi.walletBalance(walletId, 1))
+}
+
+async function sendCoins(account, address, cryptoAtoms, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId, index }) =>
+      cardanoApi.sendCoins(walletId, index, address, cryptoAtoms)
+    )
+}
+
+async function newAddress(account, info) {
+  return checkCryptoCode(info.cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId, index }) => cardanoApi.newAddress(walletId, index))
+}
+
+async function getStatus(account, toAddress, requested, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() =>
+      Promise.all([
+        cardanoApi.addressBalance(toAddress, 1),
+        cardanoApi.addressBalance(toAddress, 0)
+      ])
+    )
+    .then(([confirmedBalance, totalBalance]) => {
+      if (confirmedBalance.gte(requested)) {
+        return { status: 'confirmed' }
+      } else if (totalBalance.gte(request)) {
+        return { status: 'authorized' }
+      } else if (confirmedBalance.eq(totalBalance)) {
+        return { status: 'insufficientFunds' }
+      }
+
+      return { status: 'notSeen' }
+    })
+}
+
+async function newFunding(account, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId, index }) =>
+      Promise.all([
+        cardanoApi.walletBalance(walletId, 0),
```

Comment:
this is probably a bit redundant use of api as we could calculate both balances in one go

Reply:
maybe `walletBalance` can return total sum, and sum for min confirmed transactions - it's good solution for you?